File: backend/app/schemas/asistente.py
from enum import Enum

from pydantic import BaseModel


class RoleMensaje(str, Enum):
    user = "user"
    assistant = "assistant"


class MensajeChat(BaseModel):
    role: RoleMensaje
    content: str


class ChatRequest(BaseModel):
    mensaje: str
    historial: list[MensajeChat] = []


class ChatResponse(BaseModel):
    respuesta: str
    historial: list[MensajeChat]

File: frontend/src/types/asistente.ts
export type RoleMensaje = "user" | "assistant";

export interface MensajeChat {
  role: RoleMensaje;
  content: string;
}

export interface ChatRequest {
  mensaje: string;
  historial: MensajeChat[];
}

export interface ChatResponse {
  respuesta: string;
  historial: MensajeChat[];
}
